### ISSUE TITLE
make diagnostic.enable file type specific

### ISSUE THREAD

**Issue by laoshaw** [neoclide/coc.nvim#4267]
**Is your feature request related to a problem? Please describe.**
I used to use ALE for linting, now I turned on `diagnostic.enable:true` to use coc.nvim for linting, worked well for a few languages, however it did not work well for a language like c++, as clangd did not have as good c++20 support as g++ does, so I need use g++ to lint my c++ code.

**Describe the solution you'd like**
can turn off diagnostic.enable for certain filetypes, e.g. c++

**Describe alternatives you've considered**
back to use ALE for linting, but I miss the LSP part of coc.nvim

**Additional context**
basically i want to enable/disable diagnostic.enable per file type, for the disabled ones(e.g. c++) I will use an ALE linter.

**Comment by fannheyward** [comment]
Use:

```json
  "[cpp]": {
    "diagnostic.enable": false
  },
```

**Comment by chemzqm** [comment]
Outside languageserver, make sure use latest release branch of coc.nvim

**Comment by fannheyward** [comment]
Put in root section of coc-settings.json. You can ignore the warning by now https://github.com/neoclide/coc.nvim/discussions/4185

<img width="325" alt="截屏2022-10-08 14 36 20" src="https://user-images.githubusercontent.com/345274/194693578-a904f3b1-c8c9-44d7-81a7-cabbb966685e.png">
